dev/diffs/3.4.2.diff

```diff
@@ -976,25 +980,18 @@ index 266bb343526..85ec36db996 100644
  
            val bucketColumnType = bucketedDataFrame.schema.apply(bucketColumnIndex).dataType
            val rowsWithInvalidBuckets = fileScan.execute().filter(row => {
-@@ -461,18 +471,29 @@ abstract class BucketedReadSuite extends QueryTest with SQLTestUtils with Adapti
+@@ -461,18 +472,22 @@ abstract class BucketedReadSuite extends QueryTest with SQLTestUtils with Adapti
  
          // check existence of shuffle
          assert(
 -          joinOperator.left.exists(_.isInstanceOf[ShuffleExchangeExec]) == shuffleLeft,
-+          joinOperator.left.find { op =>
-+            op.isInstanceOf[SortExec] ||
-+              (op.isInstanceOf[CometExec] &&
-+                op.asInstanceOf[CometExec].originalPlan.find(_.isInstanceOf[SortExec]).isDefined)
-+          }.isDefined == sortLeft,
-+
++          joinOperator.left.exists(op => op.isInstanceOf[ShuffleExchangeExec] ||
```

Comment:
here's another change

Reply:
For `op.isInstanceOf[ShuffleExchangeExec]`, don't we need to check if it equals to `shuffleLeft`?

Reply:
hmm? it is checked in the code: 

```scala
exists(op => op.isInstanceOf[ShuffleExchangeExec] || op.isInstanceOf[CometShuffleExchangeExec]) == shuffleLeft

Reply:
Oh, yea, I read it wrongly.